source/python/idetect/scraper.py

```diff
@@ -90,107 +116,79 @@ def format_date(date_string):
-        else:  # Temporary fix to deal with https://github.com/codelucas/newspaper/issues/280
-            return "retrieval_failed", None, "", datetime.datetime.now(), "", ""
-
-    def get_pdf(self, url):
-        ''' Takes a pdf url, downloads it and saves it locally.'''
+def html_article(article, session):
+    """Downloads and extracts content plus metadata for html page
+    Parameters
+    ----------
+    article: article object to be scraped
+    session: the article session
+
+    Returns
+    -------
+    article: The updated article object
+    """
+
+    a = newspaper.Article(article.url)
+    a.download()
+    if a.download_state == 2:
+        a.parse()
+        article.domain = a.source_url
+        article.title = a.title
+        article.authors = a.authors
+        article.publication_date = a.publish_date
+
+        content = Content(article=[article],
+                          content=remove_newline(a.text), content_type='text')
+        session.add(content)
+        session.commit()
+
+        return article
+    else:  # Temporary fix to deal with https://github.com/codelucas/newspaper/issues/280
+        raise Exception("Retrieval Failed")
+
+
+def get_pdf(url):
+    ''' Takes a pdf url, downloads it and saves it locally.'''
+    pdf_file = NamedTemporaryFile( suffix=".pdf", prefix="tmp_", delete=False )
+    response = request.urlopen(url)  # not sure if this is needed?
+    publish_date = response.getheader('Last-Modified')
+    #pdf_file = open(ntf, 'wb')
+    pdf_file.write(response.read())
+    pdf_file.close()
+    return os.path.join('./', pdf_file.name), publish_date
+
+
+def get_body_text(url):
+    ''' This function will extract all text from the url passed in
+    '''
+    filepath, publish_date = get_pdf(url)
+    print(filepath)
+    if filepath == '':
+        return '', None
+    else:
         try:
-            response = request.urlopen(url)  # not sure if this is needed?
-            publish_date = response.getheader('Last-Modified')
-            pdf_file = open('file_to_convert.pdf', 'wb')
-            pdf_file.write(response.read())
-            pdf_file.close()
-            return os.path.join('./', 'file_to_convert.pdf'), publish_date
-        except (urllib.error.HTTPError, urllib.error.URLError, UnicodeEncodeError, ValueError) as e:
-            return '', ''
-
-    def get_body_text(self, url):
-        ''' This function will extract all text from the url passed in
-        '''
-        filepath, publish_date = self.get_pdf(url)
-        if filepath == '':
-            return '', None
-        else:
             text = str(textract.process(filepath, method='pdfminer'), 'utf-8')
             text = text.replace('\n', ' ')  # can replace with a call to
             text = text.replace('\xa0', ' ')  # the helper function.
             publish_date = format_date(publish_date)
+            os.unlink(filepath)
```

Comment:
I would personally put this in a `finally` clause rather than doing it in both the `try` and `except` clauses. Functionally the same, but a tiny bit less repetition.